### PR TITLE
Laboratory work #4, Menyalshchikov Timur - 21FPL1

### PR DESCRIPTION
**Pull request by Timur20033**

### CODE REVIEW

**Review by artyomtugaryov — APPROVED** [review]
What about 10?

File: lab_4_summarization_textrank/main.py [hunk 1]
```diff
@@ -84,59 +104,99 @@ def __init__(self, stop_words: tuple[str, ...], punctuation: tuple[str, ...]) ->
...
+        text = text.replace('\n', ' ').replace("  ", " ")
+        dogs = ""
+        for idx, symbol in enumerate(text[:-2]):
+            if symbol in ".?!" and text[idx + 1] == ' ' and text[idx + 2].isupper():
+                dogs += symbol
+                dogs += "@"
+            else:
+                dogs += symbol
+        dogs += text[-2:]
+        sentences = dogs.split("@")
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:136]: improve

File: lab_4_summarization_textrank/main.py [hunk 2]
```diff
@@ -84,59 +104,99 @@ def __init__(self, stop_words: tuple[str, ...], punctuation: tuple[str, ...]) ->
...
         :param tokens: a sequence of string tokens
         :return:
         """
-        pass
+        if not isinstance(tokens, tuple):
+            raise ValueError
+        if not all(isinstance(token, str) for token in tokens):
+            raise ValueError
+        new_tokens = (token for token in tokens if token not in self._word2id)
+        for idx, token in enumerate(new_tokens, 1000 + len(self._word2id)):
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:183]: own instance attribute - 1000

File: lab_4_summarization_textrank/main.py [hunk 3]
```diff
@@ -160,22 +226,29 @@ def __init__(self) -> None:
...
     def calculate_inout_score(self, vertex: Sentence) -> int:
         """
         Retrieves a number of vertices that are similar (i.e. have similarity score > 0) to the input one
         :param vertex
         :return:
         """
-        pass
+        if not isinstance(vertex, Sentence):
+            raise ValueError
+        summa = 0
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:247]: normal name

**Review by demid5111 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**